### ISSUE TITLE
Ditributed training problem.

### ISSUE THREAD

**Issue by YvanYin** [mit-han-lab/e3d#5]
Hi,
I faced a distributed training problem. 
I run the training script:
`torchpack dist-run -np 4 python train.py configs/semantic_kitti/spvcnn/cr0p5.yaml
`
I found the following problems:
1) I used 4 gpus, and change the batchsize from 2 to 4 in this config.
 I found that only 1 gpu is used 100%, while others are not used. (see the following Figure)
![image](https://user-images.githubusercontent.com/44796023/93211860-f70eed00-f7a0-11ea-8ea3-a845315294c7.png)

Besides, when I trained several steps, the code was stuck. There are no errors, just stuck.  I have tried many times. Some logs are illustraed as follows. 
`[loss] = 0.365:   5% 252/5398 [07:05<2:21:57,  1.66s/it]
[loss] = 0.365:   5% 253/5398 [07:05<2:17:34,  1.60s/it]
[loss] = 0.493:   5% 253/5398 [07:07<2:17:34,  1.60s/it]
[loss] = 0.493:   5% 254/5398 [07:07<2:22:29,  1.66s/it]
[loss] = 0.582:   5% 254/5398 [07:08<2:22:29,  1.66s/it]
[loss] = 0.582:   5% 255/5398 [07:08<2:20:24,  1.64s/it]
[loss] = 0.0939:   5% 255/5398 [07:10<2:20:24,  1.64s/it]
[loss] = 0.0939:   5% 256/5398 [07:10<2:19:29,  1.63s/it]
[loss] = 0.291:   5% 256/5398 [07:11<2:19:29,  1.63s/it] 
[loss] = 0.291:   5% 257/5398 [07:11<2:14:54,  1.57s/it]
[loss] = 0.225:   5% 257/5398 [07:13<2:14:54,  1.57s/it]
[loss] = 0.225:   5% 258/5398 [07:13<2:15:47,  1.59s/it]`

All my configs are as follows. 
`workers_per_gpu: 4
data:
  num_classes: 1
  ignore_label: 255
  training_size: 19132
train:
  seed: 1588147245
  deterministic: False
dataset:
  name: shift_prediction
  root: ../../data/Scannet
  num_points: 80000
  voxel_size: 0.01
num_epochs: 200
batch_size: 4
criterion:
  name: regression
optimizer:
  name: sgd
  lr: 0.24
  weight_decay: 0.0001
  momentum: 0.9
  nesterov: True
scheduler:
  name: cosine_warmup
model:
  name: spvcnn_classification
  cr: 1.0
  input_channel: 3`

Looking forward to your reply. 
Thx a lot.

**Comment by kentangSJTU** [comment]
Hi,

I haven't experienced with stucks in my experiments. Could you please provide more details? For example:

- does the problem exist for other models such as MinkowskiNet?
- can you run evaluation code normally?
- did you set some flags like CUDA_VISIBLE_DEVICES=0?


Best,
Haotian

**Comment by YvanYin** [comment]
Hi,

The problem has been solved. It is a very silly problem. :).
